docker-java/src/test/java/com/github/dockerjava/cmd/StatsCmdIT.java

```diff
@@ -56,6 +57,40 @@ public void testStatsStreaming() throws InterruptedException, IOException {
 
     }
 
+    @Test
+    public void testStatsNoStreaming() throws InterruptedException, IOException {
+        TimeUnit.SECONDS.sleep(1);
+
+        CountDownLatch countDownLatch = new CountDownLatch(NUM_STATS);
+
+        String containerName = "generated_" + new SecureRandom().nextInt();
+
+        CreateContainerResponse container = dockerRule.getClient().createContainerCmd("busybox").withCmd("top")
+                .withName(containerName).exec();
+        LOG.info("Created container {}", container.toString());
+        assertThat(container.getId(), not(isEmptyString()));
+
+        dockerRule.getClient().startContainerCmd(container.getId()).exec();
+
+        StatsCallbackTest statsCallback = dockerRule.getClient().statsCmd(container.getId()).withNoStream(true).exec(
+                new StatsCallbackTest(countDownLatch));
+
+        countDownLatch.await(5, TimeUnit.SECONDS);
+        Boolean gotStats = statsCallback.gotStats();
+
+        LOG.info("Stop stats collection");
+
+        statsCallback.close();
```

Comment:
please use `try-with-resources` to avoid leaks

Reply:
Seems like all tests should be reviewed for such cases. I have tried to make them "similar"...